core/src/main/java/org/apache/iceberg/SnapshotProducer.java

```diff
@@ -197,7 +197,10 @@ protected String targetBranch() {
   }
 
   protected ExecutorService workerPool() {
-    return this.workerPool;
+    if (workerPool == null) {
+      this.workerPool = ThreadPools.getWorkerPool();
+    }
+    return workerPool;
```

Comment:
nit: new line please.
The coding guide says we need to add new line after the blocks: https://iceberg.apache.org/contribute/#block-spacing

Reply:
ack